### ISSUE TITLE
Fix 'Menu' dropdown on mobile devices

### ISSUE THREAD

**Issue by Norric1Admin** [ONSdigital/sdg-indicators#3082]
On a mobile device, the ‘hamburger’ ‘Menu’ dropdown in the main page needs to signal to screen readers when it’s opened.

**Comment by Norric1Admin** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Accessibility tickets](https://trello.com/c/E40VA79G/549-accessibility-tickets)

**Comment by SavvasStephanides** [comment]
Found an interesting presentation on SlideShare regarding making dropdown menus accessible: https://www.slideshare.net/maxdesign/creating-an-accessible-button-dropdown

**Comment by SavvasStephanides** [comment]
Pull request: https://github.com/ONSdigital/sdg-indicators/pull/3110

**Comment by Norric1Admin** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Accessibility: Sprint 12](https://trello.com/c/HZJO8O4J/576-accessibility-sprint-12)

**Comment by LucyGwilliamAdmin** [comment]
## Description

## Screenshots
**Before**

**After**


## Successfully tested on...
- [ ] Windows: IE11
- [ ] Windows: Chrome
- [ ] Windows: Firefox
- [ ] Windows: Edge
- [ ] macOS: Chrome
- [ ] macOS: Firefox
- [ ] iOS: Safari 9.3
- [ ] iOS: Chrome
- [ ] Samsung: Chrome
- [ ] Samsung: Samsung Internet

## User feedback and problems encountered...

**Comment by Norric1Admin** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Fix 'Menu' dropdown on mobile devices](https://trello.com/c/tpLkLyev/606-fix-menu-dropdown-on-mobile-devices)

**Comment by SavvasStephanides** [comment]
# Investigation updates

## Update 1

I was investigating why the expanded / collapsed status is not announced on VoiceOver for iOS Safari. I created a [CodePen](https://codepen.io/SavvStudio/full/QXKRaO) to test this. The pen contains a `<button>`, an `<li>` and a `<span>`, in order to see on which element does VoiceOver announce the expansion status on iOS. Turns out it is only announced when the element is a `<button>`. It seems that the menu option will therefore need to be changed into a button (currently it's a `span` inside an `li`), unless there's an easier workaround.

**Comment by LucyGwilliamAdmin** [comment]
@SavvasStephanides will having a button or a link instead change any other functionality? And would it be difficult to implement?

**Comment by SavvasStephanides** [comment]
@LucyGwilliamAdmin 

> will having a button or a link instead change any other functionality?

Hopefully not. I think it's a matter of creating the link or button and attaching the event for opening and closing the menu to that link/button. Easier said than done though.

> would it be difficult to implement?

It doesn't seem straight forward. Not sure how to implement it yet. I'll dig into the code and see what's the best way to do this and (fingers crossed) it won't break anything along the way 🤞

**Comment by Norric1Admin** [comment]
@SavvasStephanides Could you make it a broken link, which doesn't navigate away from the page (e.g. `<a href="">Menu</a>`), and would this read it out as a link whilst still opening the menu?

**Comment by SavvasStephanides** [comment]
@Norric1Admin That's what I'm considering. Turning the `<span>` into a link that takes the responsibility of expanding and collapsing the menu. I'll carrying out some tests now.

**Comment by SavvasStephanides** [comment]
# Update: 24/6/2019

I think I've made some progress. The mobile menus are now buttons. This means that the `aria-expanded` label is now annouced on mobile (test [here](https://sdg.mango-solutions.com/a11y-mobile-menu-to-button/)). The last step is to update the menu script so that it updates the `aria-expanded` label to update the screen reader when the menu is collapsed or expanded.

**Comment by Norric1Admin** [comment]
Thanks - sounds like good progress.

**Comment by SavvasStephanides** [comment]
Pull requests for `open-sdg`: https://github.com/open-sdg/open-sdg/pull/210

@Norric1Admin and @LucyGwilliamAdmin : you can carry out the tests here: https://sdg.mango-solutions.com/a11y-mobile-menu-to-button/

**Comment by LucyGwilliamAdmin** [comment]
@SavvasStephanides Is this working on all devices now do you know?

**Comment by LucyGwilliamAdmin** [comment]
When I use Android browser and navigate to menu, screen reader reads:
Menu, navigation in list level one, two items. Double tap to activate.
Then when I activate Goals (first item in list is highlighted)

Is this right?

**Comment by Norric1Admin** [comment]
This failed on Samsung, but I made this clear in the report and DAC decided to pass this it AA standard. This is one to come back to if we want the 'AAA' certification.

**Comment by AnnCorp** [comment]
Resource focussed on achieving AA, so won't be actioning AAA fixes currently.
